Replace Maint. Drone machinery whitelist with last touched check & other tweaks

JJRcop opened the pull request:
## About The Pull Request

@ninjanomnom came up with the idea to replace the machine whitelist on drones with a restriction against machines used in the last 5 minutes by another being. I asked the headmins what they thought of the idea, and they gave their approval.

Air Alarms and most Computers are still blocked due to being high risk abuse vectors, but the Chief Engineer can enable them on his PDA. This also has headmin approval.

I've also done a little tidying up around some of the other rough edges with drones

 * Fixed drone minds being named "drone" instead of "Drone (563)"
     * This fix applies to any corpse with unique name enabled
 * Drones can use chisels and wallframes
 * Simple robots like the cleanbot and floorbot will not restrict drones anymore
 * Drones now respect the USE_EXP_RESTRICTIONS_ADMIN_BYPASS config var

Here are some additional changes in the pull request not related to drones

* `/atom/proc/tool_act` returns a bitflag instead of TRUE
* Air alarm now properly uses `tooltype_act` instead of the legacy `attackby` override

## Why It's Good For The Game
Opens up many more interactions for drones when it comes to constructing and maintaining different machines. 

## Changelog
:cl: JJRcop, ninjanomnom
qol: Instead of a machinery whitelist, Drones cannot interact with machines which were touched by a being in the past 5 minutes.
qol: Drones have a new machinery blacklist which currently contains the Air Alarm, and most Computers. The CE can disable this on their PDA.
qol: Drones aren't blocked by simple robots anymore. (Cleanbot, Floorbot, etc)
qol: Drones can now use chisels, and construct APCs, newscasters, lights, and other things placed on walls.
fix: Drones now respect USE_EXP_RESTRICTIONS_ADMIN_BYPASS
/:cl:

JJRcop committed:
Add technoshy component and add to Maint Drones
The technoshy component blocks machine interactions when the machine was touched
by another mob not in a whitelist within a set time. Drones were configured to
use the last 5 minutes.

JJRcop committed:
Add COMSIG_TRY_WIRES_INTERACT signal, implement in technoshy

JJRcop committed:
Allow Maint Drones to hold chisels and wallframes (buttons, firealarms, apcs, …)

JJRcop committed:
Drones don't consider simple robots as beings
Like cleanbot, medbot, beepsky, etc

JJRcop committed:
Fixes ghost corpse mind name with unique_name

JJRcop committed:
Fix COMSIG_ATOM_TOOL_ACT properly blocking tooltype_acts

JJRcop committed:
Air alarms properly uses tooltype_act

JJRcop committed:
Move COMSIG_TRY_WIRES_INTERACT into /datum/wires/interactable

JJRcop committed:
Require /datum/wires/interactable to call parent to check for signal

JJRcop committed:
Fix air alarm deconstruction state machine

ATH1909 commented on code/datums/wires/explosive.dm:
```diff
@@ -66,6 +68,8 @@
 	holder_type = /obj/item/grenade/c4
 
 /datum/wires/explosive/c4/interactable(mob/user) // No need to unscrew wire panels on plastic explosives
+	if(!..())
+		return TRUE
 	return TRUE
```
????

JJRcop replied:
Haha, I did find and replace, so this happened. The parent proc returns true anyway, so this definition is useless

JJRcop replied:
I don't even know man... What exactly was I thinking?

ATH1909 commented on code/modules/mob/living/simple_animal/friendly/drone/_drone.dm:
```diff
@@ -366,19 +354,19 @@
 	var/list/drone_bad_areas = make_associative(drone_area_blacklist_flat) + typecacheof(drone_area_blacklist_recursive)
-	var/list/drone_good_machines = make_associative(drone_machinery_whitelist_flat) + typecacheof(drone_machinery_whitelist_recursive)
 	var/list/drone_good_items = make_associative(drone_item_whitelist_flat) + typecacheof(drone_item_whitelist_recursive)
+	var/static/list/not_shy_of = typecacheof(list(/mob/living/simple_animal/drone, /mob/living/simple_animal/bot))
```
does this mean that drones aren't shy of pAI-controlled bots or of wizards who have shapeshifted into ED-209s?

JJRcop replied:
that's right. i think it's too much of an edge case to worry about

JJRcop committed:
Remove C4 interactable proc
Duplicated parent proc with no functional additions

JJRcop committed:
Move some PDA defines to new file __DEFINES/devices.dm

JJRcop committed:
Drones can't use air alarms
The Chief Engineer has an option on their PDA to disable this blacklist

JJRcop commented:
Received headmin approval for new machinery blacklist, with CE gaining the ability to disable this.

JJRcop commented:
Proposal: Add exosuit fabricator to blacklist so drones can't print more drones. Thoughts?

JJRcop committed:
Drones respect USE_EXP_RESTRICTIONS_ADMIN_BYPASS

JJRcop committed:
Drones are blacklisted from computers
They open up abuse vectors
Like Air Alarms the CE can allow this through the Drone Blacklist button
on their PDA

JJRcop committed:
Add notice to drones about Drone Chat

JJRcop commented:
> Proposal: Add exosuit fabricator to blacklist so drones can't print more drones. Thoughts?
I've decided no, but I did add computers to the blacklist

JJRcop committed:
Drones respect DB_FLAG_EXEMPT now

JJRcop commented:
Coming up before undraft: drones can use certain computers
ETA: later this afternoon

JJRcop committed:
Drones have a machinery whitelist
This cancels out any machinery on it that's also in the blacklist

JJRcop commented:
Whitelisted arcades, and some computers.

JJRcop commented on code/datums/components/technointrovert.dm:
```diff
@@ -29,8 +30,19 @@
 
+/datum/component/technointrovert/proc/on_try_wires_interact(datum/source, atom/machine)
+	SIGNAL_HANDLER
+	if(!ismachinery(machine))
+		return
+	else if(!is_in_whitelist(source, machine))
+		return COMPONENT_CANT_INTERACT_WIRES
```
I feel pretty silly now

JJRcop commented on code/_onclick/item_attack.dm:
```diff
@@ -11,7 +11,7 @@
 	var/is_right_clicking = LAZYACCESS(params2list(params), RIGHT_CLICK)
 
-	if(tool_behaviour && target.tool_act(user, src, tool_behaviour, is_right_clicking))
+	if(tool_behaviour && (target.tool_act(user, src, tool_behaviour, is_right_clicking) & TOOL_ACT_MELEE_CHAIN_BLOCKING))
```
This is the only consumer

JJRcop committed:
Formatting changes for technointrovert and drone_machine_blacklist_enabled

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

JJRcop commented:
Requested changes addressed.
